### PR TITLE
graph_rewrite sort src

### PR DESCRIPTION
**Pull request by DKormann**
improves matching and almost solves symbolic equality

### CODE REVIEW

File: test/test_dtype.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def _assert_eq(tensor:Tensor, target_dtype:DType, target):
   if DEBUG >= 2: print(tensor.numpy())
   try:
     assert tensor.dtype == target_dtype
-    np.testing.assert_allclose(tensor.numpy(), target, rtol={dtypes.float16:1e-3, dtypes.bfloat16:1e-2}.get(target_dtype, 1e-7))
+    np.testing.assert_allclose(tensor.numpy(), target, rtol={dtypes.float16:1e-3, dtypes.bfloat16:1e-2}.get(target_dtype, 1e-6))
```
Comment (chenyuxyz) [inline test/test_dtype.py:38]: why is this needed?

File: test/test_uop_graph.py [hunk 2]
```diff
@@ -15,8 +15,8 @@
 
 class TestGraphRewrite(unittest.TestCase):
   def test_dedup(self):
-    v1 = UOp(UOps.DEFINE_VAR, dtypes.float)
-    v2 = UOp(UOps.DEFINE_VAR, dtypes.float)
+    v1 = UOp(UOps.DEFINE_VAR, dtypes.float, arg=Variable('v', 0, 1))
```
Comment (chenyuxyz) [inline test/test_uop_graph.py:18]: also change this to `dtypes.int`?

File: tinygrad/codegen/uops.py [hunk 3]
```diff
@@ -350,7 +350,7 @@ def graph_rewrite(sink:UOp, pm:PatternMatcher) -> UOp:
   replace: Dict[UOp, UOp] = {}
   def __inner_rewrite(n:UOp) -> UOp:
     if n in replace: return replace[n]
-    replace_source = (n.op, n.dtype, tuple(__inner_rewrite(y) for y in n.src), n.arg)
+    replace_source = (n.op, n.dtype, (lambda x: tuple(sorted(x) if n.commutative() else x))(__inner_rewrite(y) for y in n.src), n.arg)
```
Comment (chenyuxyz) [inline tinygrad/codegen/uops.py:353]: it might be less sorts and separate the rewrite with sort if we sort in the constructor? so `n.src` here is always sorted if n is commutative, instead of only sort in rewrite

File: test/test_dtype.py (hunk 1 above)
Comment (DKormann) [reply test/test_dtype.py:38]: there are slight rounding errors due to different order of calculations 
test_dtype.py :: test_arange will produce 5.7999997 instead of 5.8

File: tinygrad/codegen/uops.py (hunk 3 above)
Comment (DKormann) [reply tinygrad/codegen/uops.py:353]: seemed like a big change but id  love to do that

File: test/test_dtype.py (hunk 1 above)
Comment (DKormann) [reply test/test_dtype.py:38]: can put it down to 3e-7
